src/server/json_family.cc

```diff
@@ -83,8 +84,173 @@ OpResult<string> OpGet(const OpArgs& op_args, string_view key, vector<pair<strin
   return out.as<string>();
 }
 
+OpResult<vector<string>> OpType(const OpArgs& op_args, string_view key, JsonExpression expression) {
+  OpResult<PrimeIterator> it_res = op_args.shard->db_slice().Find(op_args.db_ind, key, OBJ_STRING);
+  if (!it_res.ok())
+    return it_res.status();
+
+  error_code ec;
+  json_decoder<json> decoder;
+  const PrimeValue& pv = it_res.value()->second;
+
+  string val = GetString(op_args.shard, pv);
+  basic_json_parser<char> parser(basic_json_decode_options<char>{}, &JsonErrorHandler);
+
+  parser.update(val);
+  parser.finish_parse(decoder, ec);
+
+  if (!decoder.is_valid()) {
+    return OpStatus::SYNTAX_ERR;
+  }
+
+  vector<string> vec;
+  auto cb = [&vec](const string_view& path, const json& val) {
+    if (val.is_null()) {
```

Comment:
nit: I would factor out the translation logic to a separate function.
something like `string JsonType(const json& val);` and then here it will be just `vec.emplace_back(JsonType(val));`